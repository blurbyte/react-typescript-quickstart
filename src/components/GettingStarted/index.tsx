import * as React from 'react';

import SecondaryHeadline from '../SecondaryHeadline';
import Text from '../Text';
import TextList from '../TextList';
import ExternalLink from '../ExternalLink';

class GettingStarted extends React.Component {
  shouldComponentUpdate() {
    return false;
  }

  render() {
    return (
      <section>
        <SecondaryHeadline>Getting started!</SecondaryHeadline>
        <Text>Just follow a few easy steps to set up a project:</Text>
        <TextList ordered>
          <TextList.Item>
            Install <strong>Node > 8</strong>.
          </TextList.Item>
          <TextList.Item>
            Install <strong>Git</strong>.
          </TextList.Item>
          <TextList.Item>
            Clone the repository{' '}
            <code>
              git clone https:
              {'//'}
              github.com/blurbyte/react-typescript-quickstart
            </code>
            .
          </TextList.Item>
          <TextList.Item>
            Install <em>React Developer Tools</em> &mdash; provides insight about virtual DOM and components props and
            state.
          </TextList.Item>
          <TextList.Item>
            Enter project folder and install all required modules with simple <code>npm install</code> command.
          </TextList.Item>
          <TextList.Item>
            Start demo app in <em>development</em> mode <code>npm start</code>. It will build a project, start webserver
            and open app in your browser ðŸ™‚.
          </TextList.Item>
          <TextList.Item>
            All project files are placed under <strong>/src</strong> folder.
          </TextList.Item>
          <TextList.Item>
            To make a <em>production</em> build run <code>npm run build</code>, which will create ready to deploy static
            files in <strong>/dist</strong> directory.
          </TextList.Item>
        </TextList>
        <Text>
          Besides that app is actually live at{' '}
          <ExternalLink href="https://reactquickstart.surge.sh">reactquickstart.surge.sh</ExternalLink> ðŸš€.
        </Text>
      </section>
    );
  }
}

export default GettingStarted;
