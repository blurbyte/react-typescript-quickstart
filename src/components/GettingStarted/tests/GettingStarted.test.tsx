import * as React from 'react';
import { shallow } from 'enzyme';

import GettingStarted from '../index';

describe('CoreConcepts', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<GettingStarted />);
    expect(wrapper).toMatchSnapshot();
  });
});
