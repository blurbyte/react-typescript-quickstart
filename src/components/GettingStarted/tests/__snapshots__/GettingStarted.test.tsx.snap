// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoreConcepts renders correctly 1`] = `
<section>
  <styled.h2>
    Getting started!
  </styled.h2>
  <styled.p>
    Just follow a few easy steps to set up a project:
  </styled.p>
  <TextList
    ordered={true}
  >
    <Styled(styled.li)>
      Install 
      <strong>
        Node &gt; 8
      </strong>
      .
    </Styled(styled.li)>
    <Styled(styled.li)>
      Install 
      <strong>
        Git
      </strong>
      .
    </Styled(styled.li)>
    <Styled(styled.li)>
      Clone the repository 
      <code>
        git clone https:
        //
        github.com/blurbyte/react-typescript-quickstart
      </code>
      .
    </Styled(styled.li)>
    <Styled(styled.li)>
      Install 
      <em>
        React Developer Tools
      </em>
       â€” provides insight about virtual DOM and components props and state.
    </Styled(styled.li)>
    <Styled(styled.li)>
      Enter project folder and install all required modules with simple 
      <code>
        npm install
      </code>
       command.
    </Styled(styled.li)>
    <Styled(styled.li)>
      Start demo app in 
      <em>
        development
      </em>
       mode 
      <code>
        npm start
      </code>
      . It will build a project, start webserver and open app in your browser ðŸ™‚.
    </Styled(styled.li)>
    <Styled(styled.li)>
      All project files are placed under 
      <strong>
        /src
      </strong>
       folder.
    </Styled(styled.li)>
    <Styled(styled.li)>
      To make a 
      <em>
        production
      </em>
       build run 
      <code>
        npm run build
      </code>
      , which will create ready to deploy static files in 
      <strong>
        /dist
      </strong>
       directory.
    </Styled(styled.li)>
  </TextList>
  <styled.p>
    Besides that app is actually live at
     
    <styled.a
      href="https://reactquickstart.surge.sh"
    >
      reactquickstart.surge.sh
    </styled.a>
     ðŸš€.
  </styled.p>
</section>
`;
